### ISSUE TITLE
Ship binaries of NvStrapsReBar

### ISSUE THREAD

**Issue by mezj** [terminatorul/NvStrapsReBar#17]
Hi @terminatorul !
While looking through guide for NvStrapsReBar I did not notice restriction that it should be built on exactly the same machine where it is going to be added. Does it mean that there is no dependency on what computer is used for building?

If so, could you please ship prebuilt .ffs and .exe files of NvStrapsReBar, same as ReBarUEFI project did?
Thanks

**Comment by Black6spdZ** [comment]
No, it can be built on any machine.. and it takes information SPECIFIC to each setup prior to compiling the files tailored to the specific PC. Given enough time it may be streamlined to be easier

**Comment by mezj** [comment]
Just curious, what information is specific to each build? In "Building (Windows only)" guide I did not noticed any GPU or mo-bo specific information

**Comment by terminatorul** [comment]
The latest version does not use hard-coded information about the current machine (can be built anywhere).

I still have some issues to sort out before making a binary release:
-  one user reported that clearing CMOS did not recover their machine, they had to move the GPU to a different slot (replacing the GPU in the same slot would also work, if you have an extra one). I have had a similar issue on my motherboard before, but I didn't have time to do a proper test
- some users report resume from sleep gives a BSOD in Windows

so I am not in hurry to release binaries with above issues... and I do not have a lot of time for the extra development and debugging....  :(

**Comment by mezj** [comment]
@terminatorul

Thank you for answer. I understand that shipping "risky" binaries might raise number of complaints.

Still, adding paragraph to guide with proper disclaimer and link to binaries (maybe not Release section, but some web hosting site) might help people with BIOS programmers to save a lot of time to test NvStrapsReBar out :)

**Comment by Black6spdZ** [comment]
> The latest version does not use hard-coded information about the current machine (can be built anywhere).
> 
> I still have some issues to sort out before making a binary release:
> 
> * one user reported that clearing CMOS did not recover their machine, they had to move the GPU to a different slot (replacing the GPU in the same slot would also work, if you have an extra one). I have had a similar issue on my motherboard before, but I didn't have time to do a proper test
> * some users report resume from sleep gives a BSOD in Windows
> 
> so I am not ready to release binaries with above issues... and I do not have a lot of time for the extra development and debugging.... :(

so you are saying all the hardcoded stuff can be set with the NvStrapsReBar executable now??

**Comment by terminatorul** [comment]
I got to check the Clear CMOS button an pin headers on my test motherboard, (read: my safe, with QFlash Plus recovery feature, really saved me when I needed it).

And the result is I can not recover my board ... if I forget to disconnect computer from wall power, before clearing CMOS !

I can't say I understand why can't the small button just work either way ... but I guess the manufacturer tried to make sure the CMOS is under power at all possible times. Still unfriendly 'though...

I still want to implement a failsafe feature, so people don't have to open the case, maybe pull out the GPU to access the baterry, and clear CMOS... But it's not a priority now

**Comment by terminatorul** [comment]
Now my board broke again ... for no reason. Clear CMOS button + pin header not recovering the board.  It all started with unexplained WinAPI errors trying to write the EFI variable (the message only says "due to an I/O error" and the screen flickered...)

**Comment by terminatorul** [comment]
https://github.com/terminatorul/NvStrapsReBar/releases/tag/v0.2

**Comment by xCuri0** [comment]
Readme still looks like you have to manually build it, might want to change it a bit.

For ReBarUEFI I put build instructions in the wiki here can put in BUILDING.md I guess

**Comment by terminatorul** [comment]
I moved to wiki right away, maybe you just need a refresh

**Comment by unoMato** [comment]
Windows defender detecs Trojan:Win32/Wacatac.B!ml in NvStrapsReBar.exe

**Comment by terminatorul** [comment]
@unoMato

I ran a quick scan of my computer using Microsoft Safety Scan, including NvStrapsReBar.exe file, it says it's ok ....
![image](https://github.com/terminatorul/NvStrapsReBar/assets/378924/4c2e5154-645a-49c4-b5c2-ab503eb73e8d)

So I don't know what to say, Could you please also download Microsoft Safety Scan an runt the quick scan on your computer ?

I will run a full scan later. I also noticed 3 out of 72 scanners on VirusTotal say they detect something.

I recommend building from source. All source code in the executable has been written or re-written by me and there is nothing malicious in it.

**Comment by terminatorul** [comment]
I suppose automating the build with github actions would help deal with such problems as well, but the documentation and tutorials could fill a book, and I just can't deal with that now...

**Comment by unoMato** [comment]
@terminatorul 

Sorry to keep you waiting, I wasn't at home for couple of days.

It most likely was false positive and maybe MS fixed it in later database update.  When I started to download NvStrapsReBar.exe Defender said thread detected and removed it instantly. Now after couple of days I can download that file without issue and no threats are detected.

**Comment by terminatorul** [comment]
Thank you, I was worried for this !
